File: data/multilang_comparison/program_julia/src/func51.jl
include("program_julia_head.jl")
function func51(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array38 = nothing
   if pCounter > 1
       array38 = vars.data[pCounter]
       pCounter -= 1
       array38.refC += 1
   else
       array38 = JArray(zeros(Int, 683), 683, 1, 38)
   end
   loopLimit13 = (50)/4 + 1
   for i in 1:loopLimit13 
      for i in 0:(array38.size-1) 
         array38.data[i+1] += 1
      end
   end
   for i in 1:array38.size
       if array38.data[i] == 28
               return array38
      end
   end
   return array38;
end

File: data/multilang_comparison/program_cpp_array/src/func51.cpp
#include "program_cpp_array.hpp" 
Array* func51(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array38;
   if (pCounter > 0) {
      array38 = vars->data[--pCounter];
      array38->refC++;
      DEBUG_COPY(array38->id);
   } else {
      array38 = new Array();
      array38->size = 683;
      array38->refC = 1;
      array38->id = 38;
      array38->data = new unsigned int[array38->size]();
      DEBUG_NEW(array38->id);
   }
   unsigned int loop13 = 0;
   unsigned int loopLimit13 = (50)/4 + 1;
   for(; loop13 < loopLimit13; loop13++) {
      for (int i = 0; i < array38->size; i++) {
         array38->data[i]++;
      }
   }
   for (int i = 0; i < array38->size; i++) {
       if (array38->data[i] == 28) {
           return array38;
       }
   }
   return array38;
}

